rest/codegen/server.go

```diff
@@ -1,11 +1,288 @@
 package codegen
 
 import (
+	"fmt"
+	"text/template"
+
 	"goa.design/goa.v2/codegen"
+	goadesign "goa.design/goa.v2/design"
 	"goa.design/goa.v2/rest/design"
+	"net/http"
 )
 
+//todo: in encode headers and mappings
+//todo: in decode payload, query and paths
+
+const serverT = `
```

Comment:
Just curious: what's the benefit of doing these with inline defines vs. different string constants? If we are going to do it this way then we should do it this way in all writers.